corehq/middleware.py

```diff
@@ -284,3 +311,17 @@ def process_request(self, request):
         if self._bypass_sessions(request):
             request.session.save = lambda *x: None
             request._bypass_sessions = True
+
+
+def get_view_func(view_fn, view_kwargs):
+    """Given a view_fn from the `process_view` middleware function return the actual
+    function or class that represents the view.
+    """
+    if getattr(view_fn, 'is_hq_report', False):  # HQ report
+        dispatcher = view_fn.view_class
+        return dispatcher.get_report(view_kwargs['domain'], view_kwargs['report_slug'])
```

Comment:
@snopoke this is causing an issue on staging right now for Admin reports because domain is not present for those specific reports. I'm going to revert this so that it doesn't get deployed before this is fixed and affect accounting workflows

Reply:
unless the code is heavily feature flagged, I think it's a huge risk to deploy changes to middleware without any QA